components/ESC/src/FuelGauge.h

```diff
@@ -0,0 +1,537 @@
+
+    /**
+     * @brief Seal the BQ27441.
+     *
+     * @param[out] result: The return value.
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - (others): failed.
+     */
+    esp_err_t writeExtendedData(std::uint8_t class_id,
+                                std::uint8_t offset,
+                                std::uint8_t* data,
+                                std::uint8_t len);
+    /**
+     * @brief Read the result of a Control Command.
+     *
+     * @param[in]  function: Control command.
+     * @param[out] result: The return value.
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - (others): failed.
+     */
+    esp_err_t readControlWord(std::uint16_t function, std::uint16_t* result);
+
+    /**
+     * @brief Execute a Control Command.
+     *
+     * @param[in]  function: Control command.
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - (others): failed.
+     */
+    esp_err_t executeControlWord(std::uint16_t function);
+
+    /**
+     * @brief read a Word (std::uint16_t) from the IC.
+     *
+     * @param[in]  command: Command.
+     * @param[out] word: The return value.
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - ESP_FAIL: `word` is a null pointer, or i2c failure.
+     *      - (others): failed.
+     */
+    esp_err_t readWord(std::uint8_t command, std::uint16_t* word);
+
+    /**
+     * @brief Write bytes to I2C.
+     *
+     * @param[in] sub_addres: Sub address (command).
+     * @param[in] bytes: Byte buffer.
+     * @param[in] count: Byte buffer size.
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - ESP_FAIL: `word` is a null pointer, or i2c failure.
+     *      - (others): failed.
+     */
+    esp_err_t i2cWriteBytes(std::uint8_t sub_address,
+                            std::uint8_t* bytes,
+                            std::size_t count);
+
+    /**
+     * @brief Read bytes from the I2C.
+     *
+     * @param[in] sub_addres: Sub address (command).
+     * @param[out] bytes: Byte buffer.
+     * @param[in] count: Byte buffer size.
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - ESP_FAIL: `word` is a null pointer, or i2c failure.
+     *      - (others): failed.
+     */
+    esp_err_t i2cReadBytes(std::uint8_t sub_address,
+                           std::uint8_t* bytes,
+                           std::size_t count);
+
+    /**
+     * @brief Init i2c iface
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - (others): failed.
+     */
+    esp_err_t i2cInit();
+
+    /**
+     * @brief delete i2c iface
+     *
+     * @return
+     *      - ESP_OK: succeed.
+     *      - (others): failed.
+     */
+    esp_err_t i2cDelete();
+
+    /// I2C port configuration.
+    i2c_config_t m_conf;
+    /// Seal again the BQ27441?
+    bool m_seal_again;
+    /// Indicates if the enterConfig is in manual mode.
+    bool m_manual_config;
+};
+
+
+} // namespace esc
+
+
```

Comment:
delete extra space

Reply:
The same as above, this can be fixed in another PR.